plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1062,37 +1067,45 @@ bool chain_plugin::recover_reversible_blocks( const fc::path& db_dir, uint32_t c
 
    ilog( "Reconstructing '${reversible_dir}' from backed up reversible directory", ("reversible_dir", reversible_dir) );
 
-   chainbase::database  old_reversible( backup_dir, database::read_only, 0, true );
-   chainbase::database  new_reversible( reversible_dir, database::read_write, cache_size );
-   std::fstream         reversible_blocks;
-   reversible_blocks.open( (reversible_dir.parent_path() / std::string("portable-reversible-blocks-").append( now ) ).generic_string().c_str(),
+   optional<chainbase::database> old_reversible; 
+   try {
+	   old_reversible = chainbase::database( backup_dir, database::read_only, 0, true );
+   } catch (const std::runtime_error &) {
+	// since we are allowing for dirty, it must be incompatible
+   	ilog( "Did not recover any reversible blocks since reversible database incompatible");
+	return true;
+   }
+   	chainbase::database  new_reversible( reversible_dir, database::read_write, cache_size );
+   	std::fstream         reversible_blocks;
+   	reversible_blocks.open( (reversible_dir.parent_path() / std::string("portable-reversible-blocks-").append( now ) ).generic_string().c_str(),
                            std::ios::out | std::ios::binary );
 
-   uint32_t num = 0;
-   uint32_t start = 0;
-   uint32_t end = 0;
-   old_reversible.add_index<reversible_block_index>();
-   new_reversible.add_index<reversible_block_index>();
-   const auto& ubi = old_reversible.get_index<reversible_block_index,by_num>();
-   auto itr = ubi.begin();
-   if( itr != ubi.end() ) {
-      start = itr->blocknum;
-      end = start - 1;
-   }
-   if( truncate_at_block > 0 && start > truncate_at_block ) {
-      ilog( "Did not recover any reversible blocks since the specified block number to stop at (${stop}) is less than first block in the reversible database (${start}).", ("stop", truncate_at_block)("start", start) );
-      return true;
-   }
-   try {
-      for( ; itr != ubi.end(); ++itr ) {
-         EOS_ASSERT( itr->blocknum == end + 1, gap_in_reversible_blocks_db,
+   	uint32_t num = 0;
+   	uint32_t start = 0;
+   	uint32_t end = 0;
+   	old_reversible->add_index<reversible_block_index>();
```

Comment:
Please remove all tabs from the code.  The only diffs that should show up between 1078 and 1118 should be old_reversible using "->" instead of "."

Reply:
Evil tabs, nice catch @brianjohnson5972